src/Multipart/AbstractUploadManager.php

```diff
@@ -137,13 +137,29 @@ public function promise()
             // Complete the multipart upload.
             yield $this->execCommand('complete', $this->getCompleteParams());
             $this->state->setStatus(UploadState::COMPLETED);
-        })->otherwise(function (\Exception $e) {
-            // Throw errors from the operations as a specific Multipart error.
-            if ($e instanceof AwsException) {
-                $e = new $this->config['exception_class']($this->state, $e);
-            }
-            throw $e;
-        });
+        })->otherwise($this->buildFailureCatch());
+    }
+
+    private function transformException($e)
+    {
+        // Throw errors from the operations as a specific Multipart error.
+        if ($e instanceof AwsException) {
+            $e = new $this->config['exception_class']($this->state, $e);
+        }
+        throw $e;
+    }
+
+    private function buildFailureCatch()
+    {
+        if (PHP_MAJOR_VERSION >= 7) {
```

Comment:
This may be overkill, but I did want to propose a solution that kept the type-checking instead of just dropping it in favor of a naked `otherwise(function ($e)`

Reply:
@bradynpoulsen 

This solution makes sense to keep type-checking here.  Could you add test coverage for this change?  

Also, instead of checking ```PHP_MAJOR_VERSION``` we could check Throwable directly: 
```
if (interface_exists('Throwable')) {
```

Reply:
checking in again.